pkg/autoscaler/autoscaler/tidb_autoscaler.go

```diff
@@ -73,7 +73,15 @@ func updateTcTiDBAnnIfScale(tac *v1alpha1.TidbClusterAutoScaler) {
 	tac.Annotations[label.AnnTiDBLastAutoScalingTimestamp] = fmt.Sprintf("%d", time.Now().Unix())
 }
 
-func calculateTidbMetrics(tac *v1alpha1.TidbClusterAutoScaler, sts *appsv1.StatefulSet, client promClient.Client, instances []string) (int32, error) {
+func calculateTidbMetrics(tac *v1alpha1.TidbClusterAutoScaler, sts *appsv1.StatefulSet, instances []string) (int32, error) {
+	ep, err := genMetricsEndpoint(tac)
+	if err != nil {
+		return -1, err
+	}
+	client, err := promClient.NewClient(promClient.Config{Address: ep})
+	if err != nil {
+		return -1, err
+	}
```

Comment:
Why not get ep and initialize the client in manager.go as we do now?

Reply:
Querying Metrics from Prometheus would be the one of the data source, so I think it would be more proper to initialize Prometheus client in `calculateTidbMetrics` func instead of  `syncAutoScaling` func.

Reply:
But for one tac, you will initialize the prom client twice for tidb and tikv, i think it's not necessary.